src/core/namespace.c

```diff
@@ -318,11 +318,13 @@ static void drop_duplicates(BindMount *m, unsigned *n) {
         BindMount *f, *t, *previous;
 
         assert(m);
-        assert(n);
+
+        if (!*n)
+                return;
```

Comment:
This is a noop: if the conditions is true, the loop below runs 0 times (`f < m + *n` is never satisfied), and we assign `*n = 0` and return.

The assert should stay too: `n` should never be NULL. We dereference it anyway, so we'd get a segfault, but an explicit check is nicer.